modules/nf-core/fastqc/main.nf

```diff
@@ -25,6 +25,11 @@ process FASTQC {
     def old_new_pairs = reads instanceof Path || reads.size() == 1 ? [[ reads, "${prefix}.${reads.extension}" ]] : reads.withIndex().collect { entry, index -> [ entry, "${prefix}_${index + 1}.${entry.extension}" ] }
     def rename_to = old_new_pairs*.join(' ').join(' ')
     def renamed_files = old_new_pairs.collect{ old_name, new_name -> new_name }.join(' ')
+
+    def memory_in_mb = MemoryUnit.of("${task.memory}").toUnit('MB')
+    // FastQC memory value allowed range (100 - 10000)
+    def fastqc_memory = memory_in_mb > 10000 ? 10000 : (memory_in_mb < 100 ? 100 : memory_in_mb)
```

Comment:
Alternatively, since FastQC is the only command being run, what about simply doing this:
```suggestion
    def fastqc_memory = memory_in_mb - 100   // to account for potential container/bash overheads
```

Reply:
See previous comment, could add the `- 100 // to account for potential container/bash overheads` on top though.

Reply:
Ignore this as per the other conversation thread